### PR TITLE
Fix SHOW TABLES for hidden tpch and tpcds schemas

### LINKED ISSUE #1005: SHOW TABLES does not work for hidden tpch, tpcds schemas
`tpch."sf3000.0"` definitely exists:
```
presto> select * from tpch."sf3000.0".nation limit 1;
 nationkey |  name   | regionkey |                       comment
-----------+---------+-----------+-----------------------------------------------------
         0 | ALGERIA |         0 |  haggle. carefully final deposits detect slyly agai
(1 row)
```

but one cannot see tables in that schema:
```
presto> show tables from tpch."sf3000.0";
Query 20190615_143150_00028_gwaqu failed: line 1:1: Schema 'sf3000.0' does not exist
io.prestosql.sql.analyzer.SemanticException: line 1:1: Schema 'sf3000.0' does not exist
	at io.prestosql.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:192)
	at io.prestosql.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowTables(ShowQueriesRewrite.java:150)
	at io.prestosql.sql.tree.ShowTables.accept(ShowTables.java:72)
	at io.prestosql.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.prestosql.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java:147)
	at io.prestosql.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:53)
	at io.prestosql.sql.analyzer.Analyzer.analyze(Analyzer.java:73)
	at io.prestosql.sql.analyzer.Analyzer.analyze(Analyzer.java:68)
```

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #1005

### CODE REVIEW

File: presto-tpcds/src/main/java/io/prestosql/plugin/tpcds/TpcdsMetadata.java [hunk 1]
```diff
@@ -73,6 +73,12 @@ public TpcdsMetadata()
         this.tableNames = tableNames.build();
     }
 
+    @Override
+    public boolean schemaExists(ConnectorSession session, String schemaName)
+    {
+        return schemaNameToScaleFactor(schemaName) < 0 ? false : true;
```
Comment (findepi) [inline presto-tpcds/src/main/java/io/prestosql/plugin/tpcds/TpcdsMetadata.java]: - sf0 does not exists
- this needs to be handled the same way as in `io.prestosql.plugin.tpcds.TpcdsMetadata#getTableHandle`

```suggestion
        return schemaNameToScaleFactor(schemaName) > 0;
```

File: presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java [hunk 2]
```diff
@@ -153,6 +153,12 @@ private static StatisticsEstimator createStatisticsEstimator()
         return new StatisticsEstimator(tableStatisticsDataRepository);
     }
 
+    @Override
+    public boolean schemaExists(ConnectorSession session, String schemaName)
+    {
+        return schemaNameToScaleFactor(schemaName) < 0 ? false : true;
```
Comment (findepi) [inline presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java]: TPCH has a bug -- `io.prestosql.plugin.tpch.TpchMetadata#getTableHandle` accepts scale factor 0, and so
```
select count(*) from tpch.sf0.nation;
```
- should fail with schema not found
- but it fails with:
```
java.lang.IllegalArgumentException: Scale factor must be larger than 0
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:141)
	at io.prestosql.plugin.tpch.TpchTableHandle.<init>(TpchTableHandle.java:46)
	at io.prestosql.plugin.tpch.TpchTableHandle.<init>(TpchTableHandle.java:36)
	at io.prestosql.plugin.tpch.TpchMetadata.getTableHandle(TpchMetadata.java:176)
```

So:

- fix io.prestosql.plugin.tpch.TpchMetadata#getTableHandle's `if (scaleFactor ` as it is in tpcds
- make this line the same as in tpcds

File: presto-tpch/src/test/java/io/prestosql/plugin/tpch/TestTpchMetadata.java [hunk 3]
```diff
@@ -129,6 +130,17 @@ public void testTableStatsWithConstraints()
         });
     }
 
+    @Test
+    public void testHiddenSchemas()
```
Comment (findepi) [inline presto-tpch/src/test/java/io/prestosql/plugin/tpch/TestTpchMetadata.java]: - Add tests for tpcdsmetadata too.
- Could you please add an explicit test for `SHOW TABLES FROM ..` and maybe `USE ... ; SHOW TABLES;`?

**Comment by Praveen2112** [comment]
@findepi Thanks for your insights. Have applied the comments

File: presto-tests/src/test/java/io/prestosql/tests/TestTpchDistributedQueries.java [hunk 4]
```diff
@@ -116,4 +118,23 @@ public void testTableSampleSystem()
         }
         assertTrue(sampleSizeFound, "Table sample returned unexpected number of rows");
     }
+
+    public void testShowTables()
+    {
+        Session session = testSessionBuilder()
```
Comment (findepi) [inline presto-tests/src/test/java/io/prestosql/tests/TestTpchDistributedQueries.java]: - inline (it's used only once)
- create a helper method that constructs a Session for given schema

File: presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java [hunk 5]
```diff
@@ -169,11 +175,11 @@ public TpchTableHandle getTableHandle(ConnectorSession session, SchemaTableName
 
         // parse the scale factor
         double scaleFactor = schemaNameToScaleFactor(tableName.getSchemaName());
-        if (scaleFactor < 0) {
-            return null;
+        if (scaleFactor > 0) {
+            return new TpchTableHandle(tableName.getTableName(), scaleFactor);
```
Comment (findepi) [inline presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java]: please revert. 
Just make the comparison:
```
if (scaleFactor <= 0) {
     return null;
}
```

File: presto-tpcds/src/main/java/io/prestosql/plugin/tpcds/TpcdsMetadata.java [hunk 6]
```diff
@@ -89,11 +95,11 @@ public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTable
 
         // parse the scale factor
         double scaleFactor = schemaNameToScaleFactor(tableName.getSchemaName());
-        if (scaleFactor <= 0) {
-            return null;
+        if (scaleFactor > 0) {
+            return new TpcdsTableHandle(tableName.getTableName(), scaleFactor);
```
Comment (findepi) [inline presto-tpcds/src/main/java/io/prestosql/plugin/tpcds/TpcdsMetadata.java]: please revert

File: presto-tpcds/src/test/java/io/prestosql/plugin/tpcds/TestTpcds.java [hunk 7]
```diff
@@ -71,4 +73,24 @@ public void testLargeInWithShortDecimal()
         assertQuerySucceeds("SELECT i_current_price FROM item WHERE i_current_price IN (i_wholesale_cost, " + longValues + ")");
         assertQuerySucceeds("SELECT i_current_price FROM item WHERE i_current_price NOT IN (i_wholesale_cost, " + longValues + ")");
     }
+
+    @Test
+    public void testShowTables()
+    {
+        Session session = testSessionBuilder()
```
Comment (findepi) [inline presto-tpcds/src/test/java/io/prestosql/plugin/tpcds/TestTpcds.java]: as in tpch case

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**